playbooks/gobig/assign-groups.yml

```diff
@@ -2,32 +2,40 @@
 
   - hosts: all
     vars:
-        hdfs_namenodes: namenodes
         hdfs_datanodes: datanodes
-        zookeepers: zookeepers
+        hdfs_namenodes: namenodes
         mesos_masters: masters
         mesos_slaves: slaves
+        spark: spark
         uvcmetrics: uvcmetrics
+        zookeepers: zookeepers
 
-        namenode_set: "{{ groups.get(hdfs_namenodes, []) }}"
         datanode_set: "{{ groups.get(hdfs_datanodes, []) }}"
+        namenode_set: "{{ groups.get(hdfs_namenodes, []) }}"
+
         hadoop_set: "{{ namenode_set | union(datanode_set) }}"
 
         master_set: "{{ groups.get(mesos_masters, []) }}"
         slave_set: "{{ groups.get(mesos_slaves, []) }}"
-        mesos_set: "{{ master_set | union(slave_set) }}"
 
-        spark_set: "{{ master_set }}"
+        mesos_set: "{{ master_set | union(slave_set) }}"
 
+        spark_set: "{{ groups.get(spark, []) | union(mesos_set) }}"
+        uvcmetrics_set: "{{ groups.get(uvcmetrics, []) }}"
```

Comment:
I'm going to have to disagree, here, since some of these sets are dependent on others.